File: Leetcode/math/69_sqrt/Solution.java
class Solution {
    public int mySqrt(int x) {
        if(x==0||x==1) return x;
        int start = 0, end = x;
        while(start<=end){
            int mid = (start+end)/2;
            if(mid<=x/mid && x/(mid+1)<(mid+1))
                return mid;
            if( mid>x/mid){
                end = mid-1;
            }else{
                start = mid+1;
            }
        }
        return -1;
    }
}

File: Leetcode/math/69_sqrt/Solution.py
class Solution(object):
    def mySqrt(self, x):
        """
        :type x: int
        :rtype: int
        """
        start, end = 0, x
        while start <= end:
            mid = (start+end)//2
            if mid**2 <= x < (mid+1)**2:
                return mid
            if mid**2 > x:
                end = mid-1
            else:
                start = mid+1
        return -1
